stock_available/models/res_config.py

```diff
@@ -31,3 +40,27 @@ class StockConfig(models.TransientModel):
              "This installs the module stock_available_mrp.\n"
              "If the module mrp is not installed, this will install it "
              "too")
+
+    stock_available_mrp_based_on = fields.Selection(
+        _get_stock_available_mrp_based_on,
+        string='based on',
+        help="Choose the field of the product which will be used to compute "
+             "potential.\nIf empty, Quantity On Hand is used.",
+    )
+
+    @api.model
+    def get_default_stock_available_mrp_based_on(self, fields):
+        res = {}
+        icp = self.env['ir.config_parameter']
+        res['stock_available_mrp_based_on'] = safe_eval(
+            icp.get_param('stock_available_mrp_based_on', 'False'))
+        if not res['stock_available_mrp_based_on']:
+            res['stock_available_mrp_based_on'] = 'qty_available'
+        return res
+
+    @api.multi
+    def set_stock_available_mrp_based_on(self):
+        if self.stock_available_mrp_based_on:
+            icp = self.env['ir.config_parameter']
+            icp.set_param('stock_available_mrp_based_on',
+                          repr(self.stock_available_mrp_based_on))
```

Comment:
Is it not because stock_available_mrp_based_on is not mandatory ? (not sure, I think it is copy/paste before all :()